### DISCUSSION TITLE
How to connect Engelmann Sensostar U wired M-Bus to tasmota

### DISCUSSION THREAD

**Discussion by naseweiss01** [arendst/Tasmota#17228] [Show and tell]
Hello,
took me a while and i found nothing in the WEB. So I had to do it myself. Unsure if this is the right place to put this, but ...

With this I read with Tasmota ([USER_CONFIG_OVERRIDE.H] changed) my Engelmann Sensostar U with wired M-Bus.

D

>B

->sensor53 r

>M 1

+1,3,rE1,0,2400,Waerme,1,10,1040FD3D16,105B005B16

1,68b3b368x23uuUUUUUU@1,Energy total,,total,0 ; Total Wärmemenge 

1,68b3b368x47uuUUUUUU@60,Flow F_akt,,F_akt,0 ; aktueller Flow in m^3

1,68b3b368x35uuUUUU@1,Power P_akt,,P_akt,0 ; aktuelle Wärme-Leistung

1,68b3b368x59uuUU@1,Temp Vorlauf,,Vorlauf,0 ; aktuelle Vorlauftemperatur

1,68b3b368x63uuUU@1,Temp Rückl,,Rückl,0 ; aktuelle Vorlauftemperatur

1,68b3b368x67uuUU@1,Temp Diff,,Diff,0 ; aktuelle Differenz Vorlauf/Rücklauftemperatur

#

-------- this I put into  User_config_override. I needed  more space to use X47, X59  --------

#ifndef _USER_CONFIG_OVERRIDE_H_

#define _USER_CONFIG_OVERRIDE_H_

#ifndef USE_SCRIPT

#define USE_SCRIPT

#endif

#ifndef USE_SML_M

#define USE_SML_M

#endif

#ifdef USE_RULES

#undef USE_RULES

#endif

#define USE_SML_SCRIPT_CMD

#define USE_SCRIPT_SUB_COMMAND

#define SML_BSIZ 160

#define MAX_METERS 3

**Comment by fruehwi** [comment]
Very nice. That is what I´m looking for. I have some Tasmota running on Sonoff´s.
It is please possible to get a wireing schematic of your solution. 
Which HW you are using? 
Thanks.
BR Tom

**Comment by naseweiss01** [comment]
https://de.aliexpress.com/item/33006188395.html?spm=a2g0o.order_list.order_list_main.5.6bb45c5fTSqeoN&gatewayAdapt=glo2deu
the Master Version !!

Wiring is easy  Vin = 5V , Vttl =3.3 - with an Wemos D1 mini , Esp8266. I use just a USB power supply.
And RX/TX "+1,**3**,rE1,0,2400,Waerme,**1**," Don't forget GND

**Comment by timo1707** [comment]
Hi @naseweiss01,

How did you found this out? i am not that familiar with this Commands...

I have the
 [METRONA heat c](https://www.engelmann.de/wp-content/uploads/2022/01/1080621007_2022-01-17_BA_S3_Metrona.pdf)
and tried to achieve the same with infrared.

since it seems that i cannot "wake" the meter up, i pressed the button and get in tasmota 
some reply which is with 0 values.
i hope i am somewhere close :)

`17:36:46.418 RSL: SENSOR = {"Time":"2023-01-04T17:36:46","Waerme":{"total":0,"F_akt":0,"P_akt":0,"Vorlauf":0,"Rückl":0,"Diff":0}}`

**Reply by naseweiss01** [reply]
How did I find out - I analysed the raw data with sensor53 d1 in the console of Tasmota. 
Waking up is a bit tricky for the infrared devices, you need to send a sequence of several 55 hex  in a different baudrate.
My wired mbus is answering without a wake up call :-)

You may want to look in this thread as well: https://www.mikrocontroller.net/topic/438972
I will cross reference this thread there, too.

This is an approach, too: https://github.com/arendst/Tasmota/discussions/17388#discussioncomment-4407236

**Comment by steinriesen** [comment]
Hello,

I’ve been using the solution from naseweis01 for about 3 years. Now the total energy exceeds 65,535 and shows now "0". I assume the telegram has changed. Any idea how to fix this?
